### ISSUE TITLE
An output format argument with no default causes rmarkdown:::default_output_format to fail with cryptic error message.

### ISSUE THREAD

**Issue by MilesMcBain** [rstudio/rmarkdown#2407]
I ran into this developing a custom output format that worked fine from the console, but threw an error only when using the ‘knit’ button in RStudio.

Upon clicking the button I see this in the console:

    Error in merge_lists(as.list(formals(format_function)), format$options,  : 
      argument "base" is missing, with no default

I found the [cause here](https://github.com/rstudio/rmarkdown/blob/main/R/output_format.R#L406) in `default_output_format()`

As you can see [here](https://github.com/MilesMcBain/issuecreep/blob/57cf4b25884f33d0953f1744aa76090bbd7112f9/R/github_issue.R#L27) the `repo` arg of my `github_issue` format had no default value.

So we get:

``` r
fake_output_format <- function(repo, number = NULL, action = "create") {
  NULL
}

fake_overlay <- list(repo = "milesmcabin/issuecreep", number = 1)

rmarkdown:::merge_lists(as.list(formals(fake_output_format)),
                        fake_overlay,
                        recursive = FALSE)
#> Error in rmarkdown:::merge_lists(as.list(formals(fake_output_format)), : argument "base" is missing, with no default

# And this is because in merge_lists:

base_list <- as.list(formals(fake_output_format))
base <- base_list[["repo"]]
is.list(base)
#> Error in eval(expr, envir, enclos): argument "base" is missing, with no default

# because base is 'missing'
base
#> Error in eval(expr, envir, enclos): argument "base" is missing, with no default
```

<sup>Created on 2022-09-02 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

I think if I add a default to my output format all should be well. This was just a little hard to figure out and I didn’t find it documented anywhere, my apologies if this is so.

Maybe this case could be detected an some kind of more useful error thrown?

PS I drafted this and submitted this issue in RStudio with my `github_issue` format. An amazing job has been done with how extensible rmarkdown output formats are! :smile:

**Comment by MilesMcBain** [comment]
Yes, this is a RStudio-only thing. VSCode-R extension doesn't call `default_output_format()` in its knit routine as far as I can see.

**Comment by juliasilge** [comment]
I ran into this same error today, with the vetiver dashboard. I also see no problem from the console, but see this error when clicking the "knit" button _and_ when deploying to RStudio Connect:

![Screen Shot 2022-10-21 at 12 49 55 PM](https://user-images.githubusercontent.com/12505835/197268545-ee239087-67b7-400c-9997-4844fb335ee5.png)

I'll try adding a default value for my custom format arguments.

**Comment by juliasilge** [comment]
That did fix it for me; the [new dashboard is now deployed here](https://colorado.rstudio.com/rsc/traffic-crash-monitor/monitor.html).

**Comment by ashne135** [comment]
j'ai un problème  dans r et voici mon code d'erreur: 
Warning: Error in <-: argument "output" is missing, with no default
  60: ui

**Comment by ashne135** [comment]
voici mon code: 
s = sample(100:500, 50)


shinyServer(function(input, output, session){
  
  # First fluid row and first infoBox  
  output$min_ <- renderInfoBox({
    infoBox(title = "Minm", 
            value = min(s),
            subtitle = "minimum value in dataset",
            fill = TRUE) 
    
  })
  
  # First fluid row and second infoBox
  output$max_ <- renderInfoBox({
    infoBox(title = "Maxm", 
            value = max(s),
            subtitle = "maximum value in dataset", fill = T, color = "yellow") 
    
  })
  
  # First fluid row and third infoBox
  output$sd_ <- renderInfoBox({
    infoBox(title = "SD", 
            value = round(sd(s), 2),
            subtitle = "Standard Deviation", icon("arrow-up")) 
    
  })
  
  # First fluid row and fourth infoBox
  output$mean_ <- renderInfoBox({
    infoBox(title = "Mean", 
            value = mean(s),
            subtitle = "Mean of dataset values",
            icon=icon("angle-double-right")) 
    
  })
  
  # second fluid row and infoBox
  output$median_ <- renderInfoBox({
    infoBox(title = tags$b("Median"), 
            value = median(s),
            subtitle= tags$i("Median of dataset"),
            icon=icon("angle-double-right")) 
    
  })
  
  # 3rd fluid row and infoBox
  output$inline <- renderInfoBox({
   infoBox(title = "Inline CSS", value = "Adjusted width and height")
  })
  
  
}
)

**Comment by cderv** [comment]
@ashne135 I don't think this is related to this issue. Please to not add to existing issue if you're not sure, better to open a new one and link to another one. Also community.rstudio.com is a better place for such question. It seems you have diffficulties to use shiny - Sharing a complete reproducible example is best. 

@ashne135 je ne pense pas que ça soit lié à cette issue. Merci de ne pas ajouter à une issue existante si pas sûr, c'est mieux d'ouvrir une nouvelle et de lié à des anciennes éventuellement. A mon avis, souci avec shiny - partager un example complet qu'on peut reproduire me semble mieux adapté si tu veux de l'aider. Aller sur community.rstudio.com est aussi une  bonne option. 

Je vais cacher les commentaires au dessus pour garer l'issue propre.
